src/IdBrokerClient.php

```diff
@@ -386,18 +386,19 @@ public function setPassword(string $employeeId, string $password)
         $statusCode = (int)$result['statusCode'];
         
         if ($statusCode !== 200) {
-            $this->reportUnexpectedResponse($result, 1490808839);
+            $this->reportUnexpectedResponse($result, 1490808839, $statusCode);
         }
     }
     
-    protected function reportUnexpectedResponse($response, $uniqueErrorCode)
+    protected function reportUnexpectedResponse($response, $uniqueErrorCode, $httpStatusCode = 0)
```

Comment:
Do you want the status code to default to `0` or `null`? I would inline towards `null` since we don't have that information in that case.

Reply:
thats fine, I followed how \Exception chooses a default error code so the behavior was the same for exception code and http status code